src/useRefs/useRefs.ts

```diff
@@ -0,0 +1,43 @@
+import { createRef, useMemo, type RefObject } from 'react';
+
+/**
+ * Utility to automatically create refs
+ *
+ * ```typescript
+ * const buttonRef = useRef<Array<HTMLButtonElement>>([]);
+ *
+ * const refs = useRef<{
+ *  heading: RefObject<HTMLHeadingElement | null>
+ *  button: RefObject<Array<HTMLButtonElement | null>>
+ * }>({
+ *   buttonRef
+ * })
+ * ```
+ *
+ * refs.heading -> ref automatically created
+ * refs.button -> ref from target object
+ */
+export function useRefs<T extends Record<string | symbol, RefObject<unknown>>>(
+  initialTarget?: Record<string | symbol, RefObject<unknown>>,
```

Comment:
If we indeed choose to not auto-create array refs in first usage, should we then at least make array-types required to pass as initial values?

Reply:
The value of a RefObject will always be `T | null`, this should not change for arrays. The arrayRef util now uses createRefArray to create an array when the ref value is `null`